File: Second Seminar/SecondSeminar/src/main/java/com/example/SecondSeminar/exception/GlobalExceptionHandler.java
package com.example.SecondSeminar.exception;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class GlobalExceptionHandler {
    @ExceptionHandler(NotFoundException.class)
    public ResponseEntity<Void> handleNotFoundException(final NotFoundException e) {
        return ResponseEntity.notFound().build();
    }

    @ExceptionHandler(IllegalAccessException.class)
    public ResponseEntity<Void> handleIllegalArgumentException(final IllegalAccessException e) {
        return ResponseEntity.badRequest().build();
    }
}

File: SeventhSeminarKotlin/src/main/kotlin/com/example/SeventhSeminarKotlin/common/exception/GlobalExceptionHandler.kt
package com.example.SeventhSeminarKotlin.common.exception

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.MethodArgumentNotValidException
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(
        MethodArgumentNotValidException::class,
    )
    fun handleMethodArgumentNotValidException(error: MethodArgumentNotValidException): ResponseEntity<ErrorResponse> {
        return ResponseEntity.badRequest()
            .body(ErrorResponse(error.message))
    }
}
